services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py

```diff
@@ -706,6 +707,17 @@ async def delete_project(request: web.Request):
                 "It cannot be deleted until the project is closed."
             )
 
+        # if the project is locked, this raises an error
+        async with projects_api.lock_with_notification(
```

Comment:
Sorry but I am not fond of this.
Here I would go with the following ideas:
1. You can get the project lock state by using [get_project_locked_state](https://github.com/ITISFoundation/osparc-simcore/blob/f3f8d57a771b3aac9c8fcadfe677ee104253c3e3/services/web/server/src/simcore_service_webserver/projects/project_lock.py), that would be cleaner than creating this weird CHECK_IF_LOCKED status
2. Another option would be to acquire the CLOSING lock here instead and pass it to the task, which would then unlock it (you can do this using an AsyncExitStack)